src/Entities/Registry.h

```diff
@@ -58,52 +52,34 @@ class Registry
 	decltype(auto) Create() { return _registry.create(); }
 	template <typename Component, typename... Args>
 	decltype(auto) Assign(entt::entity entity, [[maybe_unused]] Args&&... args) { return _registry.assign<Component>(entity, std::forward<Args>(args)...); }
+	decltype(auto) Context() { return _registry.ctx<RegistryContext>(); };
+	decltype(auto) Context() const { return _registry.ctx<RegistryContext>(); };
 	void Reset()
```

Comment:
This whole function implementation should be moved to the cpp file. Since you're refactoring, do you mind?

Reply:
Because they have deduced return types, I'll need to rework them

Reply:
Done